addons/gamification/tests/test_challenge.py

```diff
@@ -184,6 +184,66 @@ def test_30_create_challenge_with_sum_goal(self):
             "Challenge failed to start",
         )
 
+    def test_40_create_challenge_with_sum_goal(self):
+        challenge = self.env['gamification.challenge'].create({
+            'name': 'Test Challenge',
+            'state': 'draft',
+            'user_domain': '[("active", "=", True)]',
+            'reward_id': 1,
+        })
+
+        model = self.env['ir.model'].search([('model', '=', 'gamification.badge')], limit=1)
+
+        field = self.env['ir.model.fields'].search([
+            ('model', '=', 'gamification.badge'),
+            ('name', '=', 'name')
+        ], limit=1)
+
+        self.assertNotIn(field.ttype, {'integer', 'float', 'monetary'}, "Field should not be numeric")
+
+        sum_goal = self.env['gamification.goal.definition'].create({
+            'name': 'Test Definition',
+            'computation_mode': 'sum',
+            'model_id': model.id,
+            'field_id': field.id,
+        })
+
+        existing_badges_count = len(self.env['gamification.badge'].with_user(self.user_demo.id).search([]))
+        self.assertEqual(existing_badges_count, 3, f"Expected 3 existing badges, but found {existing_badges_count}")
```

Comment:
No, we don't need this one :) The purpose is to have the test independent of the exact number of pre-existing badges. We don't want to have to update the test if another badge is added in the setup.
If someone adds an existing badge in the tests setup, this test will still work because it will "automatically" be adjusted to compare 5 to 4 instead of 4 to 3.

```suggestion
```